### PR TITLE
Refactoring Hooks and Accessibility Test Suite from JS to TS

### PR DESCRIPTION
**Pull request by sharon-zhu**
### The following changes took place: 

- Hook Action Types get updated from optional to strictly typed
- Accessibility related JS files renamed to TS files 
- accTypes file updated to reflect correct interfaces 
- Accessibility components typed where necessary 
- Accessibility reducer and action type files strictly typed. 
                                   
### Files altered:
```
| src/components/ReactTestComponent/DescribeRenderer/DescribeRenderer.jsx
| src/components/TestCase/AccTestCase.jsx     
      | /AccTestComponent/ItRenderer/ItRenderer.tsx    
| src/components/TestMenu/AccTestMenu.tsx
| src/context/actions/accTestCaseActions.js          
| src/context/reducers/accTestCaseReducer.js
| src/context/reducers/hooksTestCaseReducer.ts          
| src/utils/hooksTypes.ts 
| src/utils/accTypes.ts 
```

### CODE REVIEW

**Review by taoantaoan — APPROVED** [review]
Thanks for taking this all on Sharon! Amazing.

Looks good. No major concerns. As per discussed double check the conversion of 'const describeId = Number(e.target.id);' Might also remove some of the code that is commented out.

File: src/utils/endpointTypes.ts [hunk 1]
```diff
@@ -61,50 +61,31 @@ export interface Header {
...
-
-// export interface HooksTestModalProps extends HooksTestMenuProps {
-//   isHooksModalOpen: boolean;
-//   closeHooksModal: () => void;
-// }
-
 export interface EndpointTestStatementsProps extends EndpointTestMenuProps {
   endpointStatements: Array<EndpointStatements>;
 }
+
```
Comment (annieshinn) [inline src/utils/endpointTypes.ts:70]: Could we review this file again as a team? I'm not sure what changed.

File: src/utils/accTypes.ts [hunk 2]
```diff
@@ -1,113 +1,50 @@
...
-}
 
 export interface AccTestCaseState {
   modalOpen: boolean;
-  serverFilePath: string;
-  serverFileName: string;
-  accStatements: AccObj[];
-  dbFilePath: string;
-  addDB: boolean | string;
+  describeId: number;
```
Comment (annieshinn) [inline src/utils/accTypes.ts:4]: May we code review this file again? Thank you!

**Review by annieshinn — CHANGES_REQUESTED** [review]
Hi Sharon!

Thank you so much for your hard work on this PR! I've noted some things throughout the code. I look forward to working with you to resolve it all. 😄

File: src/context/reducers/hooksTestCaseReducer.ts [hunk 3]
```diff
@@ -62,6 +62,8 @@ export const hooksTestCaseState: HooksTestCaseState = {
   hooksTestStatement: '',
   hooksStatements: [],
   statementId: 0,
+  hookFileName: '',
+  hookFilePath: '',
```
[the remark this answers is not in the document]
Comment (sharon-zhu) [reply src/context/reducers/hooksTestCaseReducer.ts:66]: Resolve TS error. The HooksTestCaseState that this object is typed to has hookFileName and hookFilePath properties. These properties used to be added when a file became associated. Now the property is initially added as an empty string.

File: src/utils/endpointTypes.ts (hunk 1 above)
Comment (sharon-zhu) [reply src/utils/endpointTypes.ts:70]: Looks to be an extra line just from moving around the file. I'll delete.

File: src/context/reducers/hooksTestCaseReducer.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (annieshinn) [reply src/context/reducers/hooksTestCaseReducer.ts:66]: Thanks for the explanation!

File: src/utils/endpointTypes.ts (hunk 1 above)
Comment (annieshinn) [reply src/utils/endpointTypes.ts:70]: Oh, I don't have issues with the line, just confused with the whole file in general.

Comment (annieshinn) [reply src/utils/endpointTypes.ts:70]: NVM thank you!

**Review by taoantaoan — APPROVED** [review]

**Review by annieshinn — APPROVED** [review]
Thanks for implementing all of these changes, Sharon. Everything looks fantastic!

**Pull request merged**